public/app/plugins/datasource/grafana-azure-monitor-datasource/types/templateVariables.ts

```diff
@@ -0,0 +1,76 @@
+export type GrafanaTemplateVariableQueryType =
+  | 'AppInsightsMetricNameQuery'
+  | 'AppInsightsGroupByQuery'
+  | 'SubscriptionsQuery'
+  | 'ResourceGroupsQuery'
+  | 'MetricDefinitionsQuery'
+  | 'ResourceNamesQuery'
+  | 'MetricNamespaceQuery'
+  | 'MetricNamesQuery'
+  | 'WorkspacesQuery'
+  | 'UnknownQuery';
+
+interface BaseGrafanaTemplateVariableQuery {
+  rawQuery?: string;
+  kind: GrafanaTemplateVariableQueryType;
```

Comment:
I don't think this `kind` property here should be needed? The specifically typed `kind` on each of the specific query types should be enough :)

Reply:
yeah I kind of went back and forth on keeping that there. The only thing it gives us really is that if you ever want to make another thing that extends this, it'll force you to add a kind property and encourage you to add to the list of types. But Idk probably more confusing than helpful! Will remove.